kernel/cClosure.ml

```diff
@@ -309,7 +309,10 @@ let is_irrelevant info r = match info.i_cache.i_mode with
 | Reduction -> false
 | Conversion -> match r with
   | Sorts.Irrelevant -> true
-  | Sorts.RelevanceVar q -> not (info.i_cache.i_sigma.qvar_relevant q)
+  | Sorts.RelevanceVar q ->
+    (* env_qualities are bound qualities so must be treated as relevant *)
+    not (Sorts.QVar.Set.mem q info.i_cache.i_env.env_qualities)
+    && not (info.i_cache.i_sigma.qvar_relevant q)
```

Comment:
This should be the responsibility of qvar_relevant to perform this check so as to centralize the code handling qualities.